### PR TITLE
[nnc] ported some more ops + added vectors to argvalue

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56766 [nnc] ported some more ops + added vectors to argvalue**

Differential Revision: [D28118331](https://our.internmc.facebook.com/intern/diff/D28118331)

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 1]
```diff
@@ -976,15 +990,167 @@ Tensor* computeFourOperand(
...
+std::pair<ScalarType, std::vector<BufHandle>> processCatList(
+    const std::vector<BufHandle>& bufList) {
+  if (bufList.size() == 0) {
+    throw std::runtime_error("Empty input list is passed to aten::cat");
+  }
+  std::vector<BufHandle> bufInputs;
+  std::vector<BufHandle> nonEmptyInputs;
+  for (auto buf : bufList) {
+    bufInputs.push_back(buf);
+    assert(buf.node()->dims().size() > 0);
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/kernel.cpp:1002]: s/assert/TORCH_INTERNAL_ASSERT/

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 2]
```diff
@@ -976,15 +990,167 @@ Tensor* computeFourOperand(
...
+    if (buf.node()->dims().size() == 1 &&
+        immediateAs<int>(buf.node()->dim(0)) == 0) {
+      continue;
+    }
+    nonEmptyInputs.push_back(buf);
+  }
+  auto maybe_dtype = bufInputs[0].dtype().scalar_type();
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/kernel.cpp:1009]: The "maybe" prefix usually indicates that the value is optional. Here it doesn't seem such.

File: torch/csrc/jit/tensorexpr/kernel.h [hunk 3]
```diff
@@ -25,14 +25,24 @@ inline std::vector<int64_t> bufferSizes(const T& t) {
 }
 using ArgNone = c10::monostate;
 using BufList = std::vector<tensorexpr::BufHandle>;
+using IntList = std::vector<int64_t>;
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/kernel.h:28]: `IntList` doesn't seem to be used anywhere.

**Review by ZolotukhinM — APPROVED** [review]
Looks good to me!

File: torch/csrc/jit/tensorexpr/kernel.h (hunk 3 above)
Comment (Chillee) [reply torch/csrc/jit/tensorexpr/kernel.h:28]: This is used in some other lowerings - I just added all the lists here.

**Pull request closed without merging**